libraries/AP_HAL_ChibiOS/Scheduler.cpp

```diff
@@ -196,11 +196,21 @@ void Scheduler::boost_end(void)
  */
 void Scheduler::delay_microseconds_boost(uint16_t usec)
 {
-    if (!_priority_boosted && in_main_thread()) {
+    const bool main_thread = in_main_thread();
+    if (!_priority_boosted && main_thread) {
         set_high_priority();
         _priority_boosted = true;
         _called_boost = true;
     }
+    if (main_thread && !hal.util->get_soft_armed() && usec >= 500) {
+        const uint32_t start_us = AP_HAL::micros();
+        call_delay_cb();
```

Comment:
I guess the assumption is that this call takes less than 500us? If it does then we spend the time doing something useful rather than sleeping. Why not do the same when armed? Just because we might overrun that 500us?